src/integrations/bluetooth-low-energy/bluetooth-low-energy.service.ts

```diff
@@ -291,12 +316,53 @@ export class BluetoothLowEnergyService
    * @param name - Name for the new device tracker
    * @returns Registered device tracker
    */
-  protected createDeviceTracker(id: string, name: string): DeviceTracker {
+  protected createDeviceTracker(
+    id: string,
+    name: string,
+    deviceCustomizations: deviceCustomizations
+  ): DeviceTracker {
+    const trackerCustomizations: Array<EntityCustomization<any>> = [
+      {
+        for: DeviceTrackerConfig,
+        overrides: {
+          device: deviceCustomizations,
+        },
+      },
+    ];
     return this.entitiesService.add(
-      new DeviceTracker(id, name, true)
+      new DeviceTracker(id, name, true),
+      trackerCustomizations
     ) as DeviceTracker;
   }
 
+  /**
+   * Creates and registers a new battery sensor.
+   *
+   * @param id - Entity ID for the new battery sensor
+   * @param name - Name for the new battery sensor
+   * @returns Registered battery sensor
+   */
+  protected createBatterySensor(
+    id: string,
+    name: string,
+    deviceCustomizations: deviceCustomizations
```

Comment:
The `deviceCustomizations` param should also be reflected in the doc comment above

Reply:
Done. Ended up renaming the parameter to `deviceInfo` as they are not really customisations but rather reference data associated with the peripheral.